canvas2D/src/Engine/babylon.canvas2dLayoutEngine.ts

```diff
@@ -201,4 +197,455 @@
             return false;
         }
     }
+
+    /**
+     * GridData is used specify what row(s) and column(s) a Primitive is placed in when its parent is using a Grid Panel Layout. 
+     */
+    export class GridData implements ILayoutData{
+
+        row:number;
+        column:number;
+        rowSpan:number;
+        columnSpan:number;
+
+        constructor(row:number, column:number, rowSpan?:number, columnSpan?:number){
+
+            this.row = row;
+            this.column = column;
+            this.rowSpan = (rowSpan == null) ? 1 : rowSpan;
+            this.columnSpan = (columnSpan == null) ? 1 : columnSpan;
+
+        }
+
+    }
+
+    class GridDimensionDefinition {
+        public static Pixels = 1;
+        public static Stars = 2;
+        public static Auto = 3;
+        _parse(value: string, res: (v: number, vp: number, t: number) => void) {
+            let v = value.toLocaleLowerCase().trim();
+            if (v.indexOf("auto") === 0) {
+                res(null, null, GridDimensionDefinition.Auto);
+            } else if (v.indexOf("*") !== -1) {
+                let i = v.indexOf("*");
+                let w = parseFloat(v.substr(0, i));
+                res(w, null, GridDimensionDefinition.Stars);
+            } else {
+                let w = parseFloat(v);
+                res(w, w, GridDimensionDefinition.Pixels);
+            }
+        }
+    }
+    class RowDefinition extends GridDimensionDefinition {
+        heightPixels: number;
+        height: number;
+        heightType: number;
+    }
+    class ColumnDefinition extends GridDimensionDefinition {
+        widthPixels: number;
+        width: number;
+        widthType: number;
+    }
+
+    @className("GridPanelLayoutEngine", "BABYLON")
+    /**
+     * A grid panel layout.  Grid panel is basically a table that has rows and columns.
+     * When adding children to a Primitive that is using grid panel layout, you must assign a GridData object to the child to indicate where the child will appear in the grid.
+     */
+    export class GridPanelLayoutEngine extends LayoutEngineBase {
+        constructor(settings: { rows: [{ height: string }], columns: [{ width: string }] }) {

```

Comment:
comment the constructor to explain how the rows and columns arrays are working. for some examples for the user to get an idea.